datadotworld/filewriter.py

```diff
@@ -0,0 +1,166 @@
+# data.world-py
+# Copyright 2017 data.world, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the
+# License.
+#
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# This product includes software developed at
+# data.world, Inc.(http://data.world/).
+from __future__ import absolute_import
+
+try:
+    from queue import Queue
+except ImportError:
+    from Queue import Queue
+from threading import Thread
+import requests
+from datadotworld.util import parse_dataset_key
+
+
+class RemoteFile:
+    def __init__(self, config, dataset_key, file_name,
+                 mode='w',
+                 timeout=None):
+        """
+        Construct a new data.world file writer - values written to the
```

Comment:
Update docs not to say file *writer*? Eventually it will read too, right? I think "file object" is the simplest concept and the one we should stick to. Update `dw.open_remote_file()` and `DataDotWorld.open_remote_file()` docstrings accordingly.